content/upgrade_labs.md

```diff
@@ -0,0 +1,410 @@
+
+### Using the VSCode Plugin
+While any editor can be used, the Chef Infra extension for Visual Studio Code features several code generators and helpful features, such as running cookstyle when recipes are saved.
+
+## Check for Data Bags
+If data bags are used, you will need a `data_bags` directory in your repo.
+Pull down the data_bags by running:
+```
+$ chef exec knife download data_bags
+```
+
+## Check for Server Searches
+Check your cookbook code for Chef Infra Server searches, which will not be possible in an Effortless context. Identify locations making search calls and replace with other mechanisms of service discovery.
+```
+$ grep 'search(' -rn cookbooks
+```
+
+## Make a Plan File
+Run:
+```
+$ hab plan init
+```
+Edit the file, `habitat/plan.sh`, (or `plan.ps1`) with the following contents:
+```
+pkg_name=<NAME FOR YOUR POLICYFILE>
+pkg_origin=<YOUR ORIGIN>
+pkg_version="0.1.0"
+pkg_maintainer="YOUR NAME AND EMAIL"
+pkg_license=("Apache-2.0")
+pkg_scaffolding="chef/scaffolding-chef-infra"
+pkg_svc_user=("root")
+scaffold_policy_name="<YOUR POLICYFILE NAME>"
+```
+
+## Update default.toml configuration
+The Effortless package will consume the data content held within habitat/default.toml for runtime options of the Chef Infra client.  While most configurations within this file are optional, `[chef_license]\acceptance` is required to be present for operation.
+
+```
+# You must accept the Chef License to use this software: https://www.chef.io/end-user-license-agreement/
+# Change [chef_license] from acceptance = "undefined" to acceptance = "accept-no-persist" if you agree to the license.
+
+[chef_license]
+acceptance = "undefined"
+
+#######################################
+# Optional settings
+#######################################
+
+# You don't usually need to change these.
+# This project provides highly tuned defaults for you.
+# If you don't have a strong reason for overriding these
+# Then it's a good idea to remove them.
+interval = 1800
+splay = 1800
+splay_first_run = 0
+run_lock_timeout = 1800
+log_level = "warn"
+env_path_prefix = "/sbin:/usr/sbin:/usr/local/sbin:/usr/local/bin:/usr/bin:/bin"
+ssl_verify_mode = ":verify_peer"
+
+[automate]
+enable = false
+server_url = "https://<automate_url>/data-collector/v0/"
+token = "<automate_token>"
+# The connection information for your Chef Automate server
+# Setting enable = false will turn off this feature
+# The server_url is the url to your Chef Automate Server
+# The token is a token with the data_collector rights to the Chef Automate API
+# Default value: false
+
+#######################################
+# End of optional settings
+#######################################
+```
+
+## Build the Package
+Before you build, make sure you have a tree that looks like:
+```
+$ tree
+├── cookbooks
+│   ├── some-cookbook/
+│   └── another-cookbook/
+├── habitat
+│   ├── default.toml
+│   └── plan.sh
+├── policyfiles
+│ └── my-policy.rb
+└── .kitchen.yml
```

Comment:
These ASCII graphics symbols fail the docs linter process. Should I use lower-end ASCII (like pipe, plus, etc?) or replace with an image, or is there another suggestion for representing directory trees?